go/consensus/tendermint/abci/errors.go

```diff
@@ -0,0 +1,32 @@
+package abci
+
+import (
+	"errors"
+	"fmt"
+)
+
+type errorUnavailableState struct {
+	inner error
+}
+
+func (e *errorUnavailableState) Error() string {
+	return fmt.Sprintf("unavailable/corrupted state: %s", e.inner.Error())
+}
+
+func (e *errorUnavailableState) Unwrap() error {
+	return e.inner
+}
+
+// UnavailableStateError wraps an error in an unavailable state error.
+func UnavailableStateError(err error) error {
+	if err == nil {
+		return nil
+	}
+	return &errorUnavailableState{err}
+}
+
+// IsUnavailableStateError returns true if any error in err's chain is an unavailable state error.
+func IsUnavailableStateError(err error) bool {
+	var e *errorUnavailableState
+	return errors.As(err, &e)
```

Comment:
Yeah, but `errors.Is` would not work because it checks for equality (e.g., all fields must be equal) while `errors.As` only requires assignment. Of course we could also implement the `Is(err error) bool` method, I'll probably do that to make it more explicit.

Reply:
oh good point